doc/doxygen/headers/glossary.h

```diff
@@ -584,8 +584,66 @@
  * template match a certain interface or behave in a certain way: such
  * constraints are called <em>concepts</em> in C++. See the discussion in
  * @ref Concepts for more information and a list of concepts in deal.II.
+ *
+ * (As a side note, concepts have been proposed as a language extension to
+ * C++ for a long time already. They would allow us to describe that a class
+ * or function has certain properties in order to be a qualified template
+ * argument. For example, it would allow us to express in C++ code that the
+ * first argument to, say, GridTools::find_closest_vertex(), must have a type
+ * that represents an actual mesh -- which we can currently only describe in
+ * words, see @ref ConceptMeshType "MeshType concept". Using C++ concepts
+ * would allow us to describe this in code and trying to call such a function
+ * with an object as first argument that is not, in fact, a mesh would yield
+ * a compiler error that makes the mismatch clear.
+ *
+ * Unfortunately, these proposals to C++ have never made it into the official
+ * C++ standards until C++20. We may start to use them once the vast majority
+ * of our users have compilers that support this standard.
+ *
+ * More information on the topic can be found at
+ * https://en.wikipedia.org/wiki/Concepts_(C%2B%2B) .
  * </dd>
  *
+ *
+ * <dt class="glossary">@anchor GlossDimension <b>Dimensions @p dim and @p spacedim</b></dt>
+ *
+ * <dd>
+ * Many classes and functions in deal.II have two template parameters,
+ * @p dim and @p spacedim. An example is the basic Triangulation class:
+ * @code
+ *   template <int dim, int spacedim=dim>
+ *   class Triangulation {...};
+ * @endcode
+ * In all of these contexts where you see @p dim and @p spacedim referenced,
+ * these arguments have the following meaning:
+ * - @p dim denotes the dimensionality of the mesh. For example, a mesh that
+ *   consists of line segments is one-dimensional and consequently corresponds
+ *   to `dim==1`. A mesh consisting of quadrilaterals then has `dim==2` and a
+ *   mesh of hexahedra has `dim==3`.
+ * - @p spacedim denotes the dimensionality of the space in which such a mesh
+ *   lives. Generally, one-dimensional meshes live in a one-dimensional space,
+ *   and similarly for two-dimensional and three-dimensional meshes that
+ *   subdivide two- and three-dimensional domains. Consequently, the
+ *   @p spacedim template argument has a default equal to @p dim. But this need
+ *   not be the case: For example, we may want to solve an equation for
+ *   sediment transport on the surface of the Earth. In this case, the domain
+ *   is the two-dimensional surface of the Earth (`dim==2`) that lives in a
+ *   three-dimensional coordinate system (`spacedim==3`).
+ * More generally, deal.II allows to solve partial differential equations on
+ * "manifolds" that are embedded in higher dimensional space. In other words,
```

Comment:
It might be better to write 'structures' here since we already use the word manifold for our class. I could go either way on this one.